aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -614,6 +614,137 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, c10::optional<int64_t> spacing_size, c10::optional<IntArrayRef> dim,  int64_t edge_order) {
+  // Helper for gradient function to make sure input data satisfies prerequisites
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 input.");
+  if (spacing_size.has_value() && !dim.has_value()) {
+    TORCH_CHECK(spacing_size.value() == 1 || spacing_size.value() == self.dim(), "torch.gradient expected spacing to be unspecified, a scalar or a list of length ", self.dim(), "but got a list of length ", spacing_size.value());
+  }
+  if (spacing_size.has_value() && dim.has_value()) {
+    TORCH_CHECK(spacing_size.value() == dim.value().size(), "torch.gradient expected its spacing and dim arguments to have the same length, but got a spacing argument of length ", spacing_size.value(), " and a dim argument of length ", dim.value().size(), "." );
+  }
+  // See discussion : https://github.com/pytorch/pytorch/issues/56036
+  TORCH_CHECK(edge_order == 1, "torch.gradient only supports edge_order=1 currently. To request support for more edge_orders please file an issue here : https://github.com/pytorch/pytorch/issues/new?assignees=&labels=&template=feature-request.md");
+  if (dim.has_value()) {
```

Comment:
The purpose of this function is to check if all input is given in correct format. So, this function is given to check how legal / illegal is dim parameter and it gives RuntimeError in case dim has illegal elements in it.

Reply:
I don't think that's the purpose of that function, although I'm not familiar with a better helper

I guess this is OK with a comment about how it's being used plus a TODO to use a different (possibly new) function in the future, too